### PR TITLE
feat(sast): add dataflow to output

### PR DESCRIPTION
**Pull request by RabeaZr**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/sast/record.py [hunk 1]
```diff
@@ -68,23 +68,48 @@ def to_string(self, compact: bool = False, use_bc_ids: bool = False) -> str:
...
-        caller_file_details = self.get_caller_file_details_string(self.caller_file_path,
-                                                                  self.caller_file_line_range)
         evaluation_message = self.get_evaluation_string(self.evaluations, self.code_block)
 
-        status_message = colored("\t{} for file - {}\n".format(status, file_details), status_color)
-
         cwe_message = colored(f'\t{self.cwe}\n') if self.cwe else ''
 
+        if self.metadata is not None and self.metadata.taint_mode is not None and self.metadata.taint_mode.data_flow \
+                is not None and len(self.metadata.taint_mode.data_flow) > 0:
```
Comment (achiar99) [inline checkov/sast/record.py:76]:
```suggestion
        if self.metadata and self.metadata.taint_mode and self.metadata.taint_mode.data_flow  and self.metadata.taint_mode.data_flow:
```

**Review by achiar99 — APPROVED** [review]

**Comment by achiar99** [comment]
@RabeaZr Can you please validate that this will work with `-o json` flag?

File: checkov/sast/record.py [hunk 2]
```diff
@@ -68,23 +68,48 @@ def to_string(self, compact: bool = False, use_bc_ids: bool = False) -> str:
...
+
         if self.check_result["result"] == CheckResult.FAILED and code_lines and not compact:
             return f"{check_message}{severity_message}{status_message}{cwe_message}{detail}{caller_file_details}{guideline_message}{code_lines}{evaluation_message}\n"
 
         if self.check_result["result"] == CheckResult.SKIPPED:
             return f"{check_message}{severity_message}{status_message}{cwe_message}{suppress_comment}{detail}{caller_file_details}{guideline_message}\n"
         else:
             return f"{check_message}{severity_message}{status_message}{cwe_message}{detail}{caller_file_details}{evaluation_message}{guideline_message}\n"
+
+    def get_code_lines_taint(self, dataflows: List[MatchLocation]) -> Tuple[str, str]:
```
Comment (omryMen) [inline checkov/sast/record.py:96]: can you add a simple test for that ?

File: checkov/sast/record.py (hunk 1 above)
Comment (pazbechor) [reply checkov/sast/record.py:76]: you wrote `self.metadata.taint_mode.data_flow  and self.metadata.taint_mode.data_flow:` twice

File: checkov/sast/record.py [hunk 3]
```diff
@@ -68,23 +68,48 @@ def to_string(self, compact: bool = False, use_bc_ids: bool = False) -> str:
...
+        code_lines = ""
+        last_file = dataflows[0].path.split('/')[-1]
+        last_line_num = dataflows[0].start.row
+        code_lines += colored("\t\t" + last_file, 'light_yellow')
+        file_details = last_file
```
Comment (pazbechor) [inline checkov/sast/record.py:101]: why special case needed? I think it can be done with one loop & reverse

Comment (RabeaZr) [reply checkov/sast/record.py:101]: because I need to know what was the first filename in order to not print the same filename multiple times

**Review by pazbechor — APPROVED** [review]

**Pull request merged**